libs/sdk-core/src/node_builder.rs

```diff
@@ -0,0 +1,29 @@
+use std::sync::Arc;
+
+use crate::backup::BackupTransport;
+#[cfg(feature = "greenlight")]
+use crate::greenlight::{GLBackupTransport, Greenlight};
+use crate::models::Config;
+use crate::node_api::{NodeAPI, NodeResult};
+use crate::persist::db::SqliteStorage;
+
+pub async fn build_node(
+    config: Config,
+    seed: Vec<u8>,
+    restore_only: Option<bool>,
+    persister: Arc<SqliteStorage>,
+) -> NodeResult<(Arc<dyn NodeAPI>, Arc<impl BackupTransport>)> {
+    #[cfg(feature = "greenlight")]
+    {
+        let greenlight = Greenlight::connect(config, seed, restore_only, persister).await?;
+        let greenlight = Arc::new(greenlight);
+        let backup_transport = GLBackupTransport {
+            inner: greenlight.clone(),
+        };
+        let backup_transport = Arc::new(backup_transport);
+        Ok((greenlight, backup_transport))
+    }
+
+    #[cfg(not(feature = "greenlight"))]
+    compile_error!("Misconfigration: one feature must be enabled (greenlight).");
```

Comment:
Instead of this compiler error, I would 
- Make the `NodeConfig` enum non_exhaustive
- go for a `match` on the `node_config`
- add a default case to the match statement where you return an error like 'unknown implementation'.